packages/eslint-plugin/tests/docs-eslint-output-snapshots/no-misused-promises.shot

```diff
@@ -66,6 +66,18 @@ eventEmitter.on('some-event', async () => {
   await doSomething();
   otherSynchronousCall();
 });
+
+interface MySyncInterface {
+  setThing(): void;
+}
+class MyClass implements MySyncInterface {
+  async setThing(): Promise<void> {
+  ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~ Promise-returning method provided where a void return was expected by heritage type 'MySyncInterface'.
+    this.thing = await fetchThing();
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+  }
+~~~
+}
```

Comment:
Couple small notes here:

1) It looks like the rule snapshot above for `EventEmitter` doesn't include an error message because it relies on `EventEmitter` being imported (which isn't included in the code sample). Should I go ahead and add the `import` to the code example and re-generate the snapshot to include the error? (I was able to do this locally, so it's just a matter of pushing that commit)
2) Took me a minute to realize what was going on with CI failing (find / look into #8497) and learn how to update this snapshot. Do we want to add a short subsection to [Local Development > Validating Changes > Tests](https://typescript-eslint.io/contributing/local-development#validating-changes) on generating snapshots with `yarn test -u` as needed?

I realize these may be out of scope so let me know what's best here :)

Reply:
For 1, #8942 - would be great if you / anybody had time to look into it!

For 2, I'm not sure 🤔. In general it's pretty standard Jest practice that snapshots get updated with `-u`.

Reply:
> 2\. Do we want to add a short subsection to [Local Development > Validating Changes > Tests](https://typescript-eslint.io/contributing/local-development#tests) on generating snapshots with `yarn test -u` as needed?

Here it is - https://github.com/typescript-eslint/typescript-eslint/issues/8997

Reply:
> For 1, https://github.com/typescript-eslint/typescript-eslint/issues/8942 - would be great if you / anybody had time to look into it!

Ah I think #8942 is a separate issue actually, about tests triggering runtime errors in rule-code and failing with unhelpful output. Vs there's no rule-code runtime error occurring here (and no tests fail); there's just no ESLint output in the snapshot for the incorrect `EventEmitter` example. But also I just saw now that you've already fixed the  example a couple days ago (#9251) so we should be all good anyway now :) (other than the broader issue of false negatives still slipping through, but that's even more outside the scope of this PR)

> Here it is - https://github.com/typescript-eslint/typescript-eslint/issues/8997

Ah nice haha. Ok I went ahead and drafted up a PR for this, let me know how it looks to you two :) #9302